pinc/Project.inc

```diff
@@ -1715,10 +1715,10 @@ class Project
      */
     public function has_entered_formatting_round()
     {
-        global $PROJECT_STATES_IN_ORDER;
-        $state_index = array_search($this->state, $PROJECT_STATES_IN_ORDER);
-        for ($i = $state_index; $i >= 0; $i--) {
-            $round = get_Round_for_project_state($PROJECT_STATES_IN_ORDER[$i]);
+        $project_states = ProjectStates::get_states();
+        $completed_states = array_slice($project_states, 0, array_search($this->state, $project_states));
```

Comment:
I think this has an off by one. The second arg to array_slice is a length, not the final index to include so, e.g.

```
$a=["A","B","C","D"];
var_dump(implode(",", array_reverse(array_slice($a, 0, array_search("C", $a)))));
```
will output `"B,A"` rather than the intended `"C,B,A"`

Reply:
Thanks for catching that!